core/descriptors.jvm/src/org/jetbrains/kotlin/load/java/JavaIncompatibilityRulesOverridabilityCondition.kt

```diff
@@ -51,72 +51,71 @@ class JavaIncompatibilityRulesOverridabilityCondition : ExternalOverridabilityCo
         return Result.UNKNOWN
     }
 
-    // This overridability condition checks two things:
-    // 1. Method accidentally having the same signature as special builtin has does not supposed to be override for it in Java class
-    // 2. In such Java class (with special signature clash) special builtin is loaded as hidden function with special signature, and
-    // it should not override non-special method in further inheritance
-    // See java.nio.Buffer
-    private fun isIncompatibleInAccordanceWithBuiltInOverridabilityRules(
-        superDescriptor: CallableDescriptor,
-        subDescriptor: CallableDescriptor,
-        subClassDescriptor: ClassDescriptor?
-    ): Boolean {
-        if (superDescriptor !is CallableMemberDescriptor || subDescriptor !is FunctionDescriptor ||
-            KotlinBuiltIns.isBuiltIn(subDescriptor)
-        ) {
-            return false
-        }
-
-        if (!subDescriptor.name.sameAsBuiltinMethodWithErasedValueParameters && !subDescriptor.name.sameAsRenamedInJvmBuiltin) {
-            return false
-        }
+    override fun getContract() = ExternalOverridabilityCondition.Contract.CONFLICTS_ONLY
 
-        val overriddenBuiltin = superDescriptor.getOverriddenSpecialBuiltin()
+    companion object {
+        // This overridability condition checks two things:
+        // 1. Method accidentally having the same signature as special builtin has does not supposed to be override for it in Java class
+        // 2. In such Java class (with special signature clash) special builtin is loaded as hidden function with special signature, and
+        // it should not override non-special method in further inheritance
+        // See java.nio.Buffer
+        fun isIncompatibleInAccordanceWithBuiltInOverridabilityRules(
```

Comment:
I don't like FIR2IR wants to use this function. May be the better solution is to write FIR version of it. Let's discuss on today meeting.